src/Makefile

```diff
@@ -1,77 +1,78 @@
 #
 # This is the main makefile for the serial version of LATTE
 #
-.SUFFIXES: .fortran90 .f90
+.SUFFIXES: .fortran90 .F90
 
 include ../makefile.CHOICES
 
+FCID = $(shell echo $(FC) | head -c 3)
+
 PPFLAGS = -DXSYEVD -D$(PRECISION)PREC -DGPU$(GPUOPT) -DDBCSR_$(DBCSR_OPT) -DMPI_$(MPIOPT)
-PPFLAGS += -DPROGRESS$(PROGRESS)
+PPFLAGS += -DPROGRESS$(PROGRESS) -DFCID$(FCID)
```

Comment:
Doesn't the IBM preprocessor set `__xlC__` as a define?

Reply:
From what I understand, yes xlf*

Reply:
Yeah, there might also be `__IBM__`.

Reply:
If you are using IBM XL C/C++ for Linux 13.1.1 or later, `__ibmxl__` is defined by default, and legacy macro `__xlC__` is defined if you compile with `-qxlcompatmacros`. If you're using an older version of the compiler, `__xlC__` is defined by default. See [this blog](https://www.ibm.com/developerworks/community/blogs/572f1638-121d-4788-8bbb-c4529577ba7d/entry/How_to_identify_the_IBM_XL_compiler_when_compiling_with_different_compilers?lang=en) for more details.